bqskit/passes/processing/treescan.py

```diff
@@ -0,0 +1,190 @@
+
+        Args:
+            start_from_left (bool): Determines where the scan starts
+                attempting to remove gates from. If True, scan goes left
+                to right, otherwise right to left. (Default: True)
+
+            success_threshold (float): The distance threshold that
+                determines successful termintation. Measured in cost
+                described by the hilbert schmidt cost function.
+                (Default: 1e-8)
+
+            cost (CostFunction | None): The cost function that determines
+                successful removal of a gate.
+                (Default: HilbertSchmidtResidualsGenerator())
+
+            instantiate_options (dict[str: Any]): Options passed directly
+                to circuit.instantiate when instantiating circuit
+                templates. (Default: {})
+
+            tree_depth (int): The depth of the tree of potential
+                solutions to instantiate. Note that 2^(tree_depth) - 1
+                circuits will be instantiated in parallel.
+
+            collection_filter (Callable[[Operation], bool] | None):
+                A predicate that determines which operations should be
+                attempted to be removed. Called with each operation
+                in the circuit. If this returns true, this pass will
+                attempt to remove that operation. Defaults to all
+                operations.
+        """
+
+        if not is_real_number(success_threshold):
+            raise TypeError(
+                'Expected real number for success_threshold'
+                ', got %s' % type(success_threshold),
+            )
+
+        if not isinstance(cost, CostFunctionGenerator):
+            raise TypeError(
+                'Expected cost to be a CostFunctionGenerator, got %s'
+                % type(cost),
+            )
+
+        if not isinstance(instantiate_options, dict):
+            raise TypeError(
+                'Expected dictionary for instantiate_options, got %s.'
+                % type(instantiate_options),
+            )
+
+        self.collection_filter = collection_filter or default_collection_filter
+
+        if not callable(self.collection_filter):
+            raise TypeError(
+                'Expected callable method that maps Operations to booleans for'
+                ' collection_filter, got %s.' % type(self.collection_filter),
+            )
+
+        self.tree_depth = tree_depth
+        self.start_from_left = start_from_left
+        self.success_threshold = success_threshold
+        self.cost = cost
+        self.instantiate_options: dict[str, Any] = {
+            'dist_tol': self.success_threshold,
+            'min_iters': 100,
+            'cost_fn_gen': self.cost,
+        }
+        self.instantiate_options.update(instantiate_options)
+
+    @staticmethod
+    def get_tree_circs(
+        orig_num_cycles: int, circuit_copy: Circuit,
+        cycle_and_ops: list[tuple[int, Operation]],
```

Comment:
To keep with the rest of the repo's style, we do one parameter per line in multi-line parameters. No need to change, but worth mentioning